src/SqlAsyncCollector.cs

```diff
@@ -95,7 +96,19 @@ public async Task AddAsync(T item, CancellationToken cancellationToken = default
                 TelemetryInstance.TrackEvent(TelemetryEventName.AddAsync);
                 try
                 {
-                    this._rows.Add(item);
+                    if (item is string)
+                    {
+                        this._rows.Add(JsonConvert.DeserializeObject<T>(item.ToString()));
```

Comment:
This may be a weird compiler thing. @MaddyDev if you remove the ToString() does the compiler complain at you?

Reply:
yes, it's a compiler thing, it's not recognizing item as string.

Reply:
Well I'm pretty confused then - the type of item is `T`, which means that the DeserializeObject here is converting a string to...a string? Or is T actually something else at this point and the parameter typing isn't correct? Because if that's the case shouldn't the type of item be `object` or something more generic?

Anyways it would be better just to cast it directly (`item as string`)- calling ToString on a string is pretty odd.